src/validator/init-validator.ts

```diff
@@ -54,8 +54,8 @@ export async function initValidator(configArg: Partial<ValidatorConfig>) {
   if (resetLedger) args.push('-r')
 
   if (programs.length > 0) {
-    args.push('--bpf-program')
     for (const { programId, deployPath } of programs) {
```

Comment:
This is incorrect as each program needs to have it's `--bpf-program` prefix if you have numerous, i.e. `--bpf-program id1 path1 --bpf-program id2 path2`

Reply:
You've just described exactly what is my PR does:

Let's say we have an array of programs
```js
const programs = [
  {
    programId: 'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s',
    deployPath: '/home/oleg/meta/amman-tmp/target/deploy/mpl_token_metadata.so',
  },
  {
    programId: 'vau1zxA2LbssAUEF7Gpw91zMM1LvXrvpzJtmZ58rPsn',
    deployPath: '/home/oleg/meta/amman-tmp/target/deploy/mpl_token_vault.so',
  },
  {
    programId: 'auctxRXPeJoc4817jDhf4HbjnhEcr1cCXenosMhK5R8',
    deployPath: '/home/oleg/meta/amman-tmp/target/deploy/mpl_auction.so',
  },
  {
    programId: 'p1exdMJcjVao65QdewkaZRUnU6VPSXhus9n2GzWfh98',
    deployPath: '/home/oleg/meta/amman-tmp/target/deploy/mpl_metaplex.so',
  },
];
```
And this code I have in PR:

```js
let args = [];

if (programs.length > 0) {

  for (const { programId, deployPath } of programs) {
    args.push('--bpf-program');
    args.push(programId);
    args.push(deployPath);

  }
}
```

The result of it would be:

```js
[
  ('--bpf-program',
  'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_token_metadata.so',
  '--bpf-program',
  'vau1zxA2LbssAUEF7Gpw91zMM1LvXrvpzJtmZ58rPsn',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_token_vault.so',
  '--bpf-program',
  'auctxRXPeJoc4817jDhf4HbjnhEcr1cCXenosMhK5R8',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_auction.so',
  '--bpf-program',
  'p1exdMJcjVao65QdewkaZRUnU6VPSXhus9n2GzWfh98',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_metaplex.so')
];
```

It's exactly what you want to expect as arguments to `solana-test-validator`

But for now we've got :

```js
if (programs.length > 0) {

  args.push('--bpf-program');

  for (const { programId, deployPath } of programs) {
    args.push(programId);
    args.push(deployPath);
  }
}
```
And result:

```js
[
  ('--bpf-program',
  'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_token_metadata.so',
  'vau1zxA2LbssAUEF7Gpw91zMM1LvXrvpzJtmZ58rPsn',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_token_vault.so',
  'auctxRXPeJoc4817jDhf4HbjnhEcr1cCXenosMhK5R8',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_auction.so',
  'p1exdMJcjVao65QdewkaZRUnU6VPSXhus9n2GzWfh98',
  '/home/oleg/meta/amman-tmp/target/deploy/mpl_metaplex.so')
];
```
 And the output is not valid, as you said earlier.

Reply:
You are totally right no this one ..